cheatsheets/Forgot_Password_Cheat_Sheet.md

```diff
@@ -1,60 +1,137 @@
-- Strength of questions used for reset should vary based on the nature of the credential. Administrator credentials should have a higher requirement.
-- The ideal implementation should rotate the questions asked in order to avoid automation.
+**Common secure practices**:
+
+- Randomly generated with a [CSPRNG](https://en.wikipedia.org/wiki/Cryptographically_secure_pseudorandom_number_generator) (*e.g.* [secrets](https://docs.python.org/3/library/secrets.html) library in Python).
+- Short lifetime (*e.g.* 30 minutes).
+- Linked to the user requesting the token in the database.
+- One time use (should be removed from the database once used).
+- Ensure that the tokens and codes are stored in a secure fashion by following the [Password Storage CS](Password_Storage_Cheat_Sheet.md) and the [Cryptographic Storage CS](Cryptographic_Storage_Cheat_Sheet.md).
+
+**Tokens secure practices**:
+
+- Long enough to avoid brute-force attacks (16 characters should be the minimum used).
+
+**Codes secure practices**:
+
+- Minimum length of 8 digits, 12 for improved security.
+- If the service allows users to view the backup codes, the codes should be [securely stored](Cryptographic_Storage_Cheat_Sheet.md) and access should only happen in an authenticated session after asking for a user identifier (password, email token, etc.).
+- A user should have multiple recovery codes at any given time to ensure that one of them works (most services provide the user with 10 backup codes).
+
+## Methods Implementation
+
+In order to implement the forgot password service, the developer needs to choose one of the proposed [methods](#methods).
+
+### One Time Password
+
+One Time Password (OTP) is the best method in order to implement a secure forgot password service that triggers as a 2FA functionality.
+
+The two most famous methods are Time-OTP ([TOTP](https://tools.ietf.org/html/rfc6238)), or HMAC-OTP ([HOTP](https://tools.ietf.org/html/rfc4226)). The main difference is in the counter, where TOTP focuses on the Unix time, and HOTP has a counter that gets incremented on every user call to generate the OTP.
+
+For a better description of OTP generation, refer to the [MFA CS](Multifactor_Authentication_Cheat_Sheet.md#something-you-have), where the pros and cons of every implementation are provided.
+
+One implementation can be found over for [Authy](https://www.twilio.com/docs/authy/tutorials/two-factor-authentication-python-flask). If you don't want to rely on applications (such as Authy, Google/Microsoft Authenticator, etc.), there are libraries like pyotp, which helps the developer implement on their own any of the methods discussed in this section.
+
+> OTPs can be sent through other channels as well, such as emails and SMSs. [Various attacks and weaknesses](https://en.wikipedia.org/wiki/SIM_swap_scam) have been identified in SMS that it is [preferrable not to use them for OTPs](https://auth0.com/blog/why-sms-multi-factor-still-matters/).
```

Comment:
Again, why is this a quote?

Reply:
Using quotes to look like a note. I will update this to follow notes that are provided by MkDocs. I'll do that after verifying the content and being happy with it.